### PR TITLE
Re-Integrate scroll wheel patches.

### PR DESCRIPTION
**Pull request by HubertHuckevoll**

### CODE REVIEW

**Comment by bluewaysw** [comment]
First of all, very good work!!
With using the "ignore whitespace" github feature the review of the diffs become usable.

**Comment by HubertHuckevoll** [comment]
> First of all, very good work!! With using the "ignore whitespace" github feature the review of the diffs become usable.

Thank you, good to know....!

**Review by bluewaysw — COMMENTED** [review]
Lets discuss what you point of view is regarding the commented parts and what might be doable.

File: CInclude/Objects/inputC.goh [hunk 1]
```diff
@@ -82,5 +83,17 @@
...
 
-@endlib
+/*
+ * Mouse Wheel message
+ * "inputState" for MSG_META_MOUSE_WHEEL_VERTICAL is wheel Info in the low byte,
+ * ShiftState in the high byte -- use GET_WHEEL_INFO and GET_SHIFT_STATE.
+ * wheel is > 0 for down and < 0 for up.
+ */
 
+@importMessage	MetaInputMessages, void MSG_META_MOUSE_WHEEL_VERTICAL(
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply CInclude/Objects/inputC.goh:93]: Okay.

File: Driver/Mouse/ctmabs/ctmabs.gp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#
+type	driver, single
+#
+# Import kernel routine definitions
+#
+library	geos
+#
+# Desktop-related things
+#
+longname 	"Basebox Mouse Emulation"
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Driver/Mouse/ctmabs/ctmabs.gp:31]: Okay, done.

File: Driver/Mouse/ctmabs/ctmabs.asm [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+
+if MOUSE_GEOS_NATIVE_WHEEL_SUPPORT
+	mouseNameTable	lptr.char	baseboxPageMouse,
+					baseboxCursorMouse,
+					baseboxNativeMouse
+			lptr.char	0	; null-terminator
+
+	baseboxPageMouse	chunk.char	'Basebox Mouse (Wheel = Page Up/Down)', 0
+	baseboxCursorMouse	chunk.char	'Basebox Mouse (Wheel = Cursor Up/Down)', 0
+	baseboxNativeMouse	chunk.char	'Basebox Mouse (Native Wheel Support)', 0
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Driver/Mouse/ctmabs/ctmabs.asm:77]: Did something along those lines.

File: Driver/Mouse/ctm/ctm.asm [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+
+if MOUSE_GEOS_NATIVE_WHEEL_SUPPORT
+	mouseNameTable	lptr.char	ctPageMouse,
+					ctCursorMouse,
+					ctNativeMouse
+			lptr.char	0	; null-terminator
+
+	ctPageMouse	chunk.char	'CTMOUSE.EXE /O (Wheel = Page Up/Down)', 0
+	ctCursorMouse	chunk.char	'CTMOUSE.EXE /O (Wheel = Cursor Up/Down)', 0
+	ctNativeMouse	chunk.char	'CTMOUSE.EXE /O (Native Wheel Support)', 0
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Driver/Mouse/ctm/ctm.asm:76]: I changed the names for the better.

File: Driver/Mouse/mouseCommon.asm [hunk 5]
```diff
@@ -1603,6 +1663,107 @@ noCombine:
 
 MouseCombineEvent endp
 
+
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Driver/Mouse/mouseCommon.asm:1666]: Did add conditional compile flags.

File: Include/Internal/mouseDr.def [hunk 6]
```diff
@@ -292,9 +292,11 @@ MouseExtendedInfo	record
 				;  "bus" mouse). This field contains the
 				;  factory-default level
     MEI_CALIBRATE:1		; Mouse can be calibrated within PC/GEOS
-    :9
+    MEI_WHEEL_VERTICAL:2	; stores the kind of wheel action that we want
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Include/Internal/mouseDr.def:295]: You're right, I removed the MEI_WHEEL_VERTICAL Flags.

File: Include/Internal/im.def [hunk 7]
```diff
@@ -475,6 +475,15 @@ MSG_IM_READ_DIGITIZER_COORDS	message
 ; Return:	nothing
 ;
 
+;-----------------------------------------------------------------------------
+;	Mouse Wheel messages
+;-----------------------------------------------------------------------------
+
+MSG_IM_MOUSE_WHEEL_VERTICAL message
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Include/Internal/im.def:482]: Well... we need to talk about htis....

File: Driver/Mouse/imps2/imps2.asm [hunk 8]
```diff
@@ -0,0 +1,602 @@
...
+
+if	MOUSE_GEOS_NATIVE_WHEEL_SUPPORT
+	mouseNameTable  lptr.char  	imps2PageMouse,
+					imps2CursorMouse,
+					imps2NativeMouse
+			lptr.char	0  ; null-terminator
+
+	imps2PageMouse    chunk.char	'IM PS/2 Wheel Mouse (Wheel = Page Up/Down)', 0
+	imps2CursorMouse  chunk.char	'IM PS/2 Wheel Mouse (Wheel = Cursor Up/Down)', 0
+	imps2NativeMouse  chunk.char	'IM PS/2 Wheel Mouse (Native Wheel Support)', 0
```
[the remark this answers is not in the document]
Comment (HubertHuckevoll) [reply Driver/Mouse/imps2/imps2.asm:51]: Changed that.

**Pull request merged**